### PR TITLE
Clean dangling calls to old Point-In-Time APIs

### LINKED ISSUE #135: [BUG] Remove references to old PIT APIs
### What is the bug?
`open_point_in_time` and `close_point_in_time` APIs are not supported by OpenSearch. See https://github.com/opensearch-project/opensearch-ruby/pull/155#discussion_r1143922457

Remove the following references: https://github.com/search?q=repo%3Aopensearch-project%2Fopensearch-rs+pit&type=code

### How can one reproduce the bug?
This client does not support these APIs, there are just some dangling references in specs.

### What is the expected behavior?
_A clear and concise description of what you expected to happen._

### What is your host/environment?
_Operating system, version._

### Do you have any screenshots?
_If applicable, add screenshots to help explain your problem._

### Do you have any additional context?
_Add any other context about the problem._

### PR DESCRIPTION
**Pull request by Swiddis**
### Description

Per #135, there are references to point in time APIs supported by ElasticSearch that we don't support. This PR:

- Removes the references to these APIs from the spec
- Removes all builders for these APIs from `opensearch/src/root`

Of note is that there is some documentation I found for [PITs in OpenSearch under `_search`](https://opensearch.org/docs/latest/search-plugins/point-in-time-api/). It may also be worth looking at implementing these structs for the alternative API rather than removing them entirely. For the moment I just took the easy way out.

### Issues Resolved

Resolves #135 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Xtansia — APPROVED** [review]
Thanks!

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
@Swiddis not against removal but may be we could refactor those to support PITs create / delete APIs instead, wdyt?

**Comment by Swiddis** [comment]
> @Swiddis not against removal but may be we could refactor those to support PITs create / delete APIs instead, wdyt?

Would probably be ideal to refactor, though I have to familiarize myself with these APIs first to see how feasible it is. Will look into it. In the meantime can I get the issue assigned to me?

**Comment by reta** [comment]
> > @Swiddis not against removal but may be we could refactor those to support PITs create / delete APIs instead, wdyt?
> 
> Would probably be ideal to refactor, though I have to familiarize myself with these APIs first to see how feasible it is. Will look into it. In the meantime can I get the issue assigned to me?

Thank you, for some reasons I cannot assign it to you, seems like the assignee is constrained ...

**Comment by Xtansia** [comment]
> > > @Swiddis not against removal but may be we could refactor those to support PITs create / delete APIs instead, wdyt?
> > 
> > 
> > Would probably be ideal to refactor, though I have to familiarize myself with these APIs first to see how feasible it is. Will look into it. In the meantime can I get the issue assigned to me?
> 
> Thank you, for some reasons I cannot assign it to you, seems like the assignee is constrained ...

You can't assign external users unless they've participated in the issue/PR (i.e. commented on it), so if @Swiddis comments on the issue saying they're working on it, then it'll be able to be assigned to them.

**Comment by Swiddis** [comment]
Updated based on my best interpretation of [the specification under `_search`](https://opensearch.org/docs/latest/search-plugins/point-in-time-api/).

File: opensearch/src/root/mod.rs [hunk 1]
```diff
@@ -438,23 +438,32 @@ where
 }
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[doc = "API parts for the Close Point In Time API"]
-pub enum ClosePointInTimeParts {
-    #[doc = "No parts"]
-    None,
+pub enum ClosePointInTimeParts<'b> {
```
Comment (reta) [inline opensearch/src/root/mod.rs:441]:
```suggestion
pub enum DeletePointInTimeParts<'b> {
```

File: opensearch/src/root/mod.rs [hunk 2]
```diff
@@ -9013,7 +9015,7 @@ impl OpenSearch {
...
+    #[doc = "[Scroll API](https://opensearch.org/docs/)\n\nAllows to retrieve a large numbers of results from a single search request."]
```
Comment (reta) [inline opensearch/src/root/mod.rs:9018]: This change seems to be unrelated, why?

File: opensearch/src/root/mod.rs [hunk 3]
```diff
@@ -5046,24 +5055,21 @@ where
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[doc = "API parts for the Open Point In Time API"]
 pub enum OpenPointInTimeParts<'b> {
-    #[doc = "No parts"]
-    None,
     #[doc = "Index"]
     Index(&'b [&'b str]),
 }
 impl<'b> OpenPointInTimeParts<'b> {
```
Comment (reta) [inline opensearch/src/root/mod.rs:5061]:
```suggestion
impl<'b> CreatePointInTimeParts<'b> {
```

File: opensearch/src/root/mod.rs (hunk 2 above)
Comment (Swiddis) [reply opensearch/src/root/mod.rs:9025]: It looks like it was a generator error? You would expect something like this if the generator logic was updated. The closest thing to that was adding `extra-traits` to `syn` because otherwise the generator fails to compile (syn traits don't implement Debug if that feature isn't enabled). Can't find a way to reproduce the old output on my machine (and the new version looks more correct?).

File: CHANGELOG.md [hunk 4]
```diff
@@ -19,6 +19,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 ### Changed
 - Updates users guide with complete examples ([#114](https://github.com/opensearch-project/opensearch-rs/pull/114))
+- References to unused Point-in-Time API updated ([#136](https://github.com/opensearch-project/opensearch-rs/pull/136))
```
[the remark this answers is not in the document]
Comment (Swiddis) [reply CHANGELOG.md]: I assume this means rewording/putting in another changelog category, does change of intent have a precise meaning?

**Comment by reta** [comment]
LGTM! Thanks a lot @Swiddis ! @Xtansia mind double checking please?
